_posts/setting-up-natty.md

```diff
@@ -0,0 +1,297 @@
+
+All of these commands will make text appear after you press enter. Unless there are errors, you shouldn't be worried.
+
+First, we run this command:
+
+    sudo apt-get install software-properties-common
+
+(for more info on that, see [What is software-properties-common](https://askubuntu.com/q/1000118/511730) on Ask Ubuntu)
+
+Then we can install the universe:
+
+    sudo add-apt-repository universe
+
+Then update:
+
+    sudo apt-get update
+
+And then finally we can install maven:
+
+    sudo apt-get install maven
+
+To check if it installed properly, you can run `mvn --version`. That should show this:
+
+Apache Maven 3.5.0
+Maven home: /usr/share/maven
+Java version: 1.8.0_151, vendor: Oracle Corporation
+Java home: /usr/lib/jvm/java-8-openjdk-amd64/jre
+Default locale: en_US, platform encoding: UTF-8
+OS name: "linux", version: "4.13.0-16-generic", arch: "amd64", family: "unix"
+
+Good! Maven is installed.
+
+#Installing Tomcat
+
+To install this simply, let's take the latest version of tomcat from [their page](http://tomcat.apache.org/download-80.cgi). This would be 8.5.29 at the moment.
+
+`wget` will retrieve the content from the link. In this case, the link is the latest tomcat version. So that's this command:
+
+    wget http://apache.mivzakim.net/tomcat/tomcat-8/v8.5.29/bin/apache-tomcat-8.5.29.tar.‌​‌​gz.
+
+Then we can unzip the file and make it usable by Natty:
+
+    tar -xzf apache-tomcat-8.5.29.tar.‌​gz
+
+And now Tomcat is set up!
+
+---
+
+Now we can start setting up Natty itself.
+
+First, since we're on a live boot, we need to install git.
+
+    sudo apt-get install git
+
+You'll be prompted to confirm.
+
+After git is installed, we can clone into the Natty repo:
+
+    git clone https://github.com/SOBotics/Natty
+
+Now the repo is cloned, but we need to get into it now! How do we do that?
+
+We need to change directories. That is, we need to be not in `ubuntu@ubuntu:~$`, but at `ubuntu@ubuntu:~/Natty$`.
+
+We can change directories using the `cd` command. You can type `cd <path>` to go to that path - for instance, if you're on `ubuntu@ubuntu:~$`, and you want to you to `ubuntu@ubuntu:~/example/path/folder$`, you can use the command `cd example/path/folder` from `ubuntu@ubuntu:~$`. Just typing `cd` and executing that will take you back to the home directory (`ubuntu@ubuntu:~$`). `cd ..` will take you from `ubuntu@ubuntu:~/example/path/folder$` to `ubuntu@ubuntu:~/example/path$`. 
+
+So we need to `cd` into `Natty`. That would be:
+
+    cd Natty
+
+Your terminal should now show `ubuntu@ubuntu:~/Natty$`.
+
+So now we need to get it back to the old revision. This is possible using the `checkout` command in git. The revision we want is located at `https://github.com/SOBotics/Natty/tree/8bc0edf614d8fbe1a04a66ab835d87bb76ced847`. So we don't need to write out that whole thing, we just need the hash (that long bit starting with 8). We don't even need the whole thing - just the beginning of it.
```

Comment:
Let's not add about the old revision in the blog. Nattyv4 is compilable and can work, but it's just that the logic needs to be updated. I asked you to use the old v3 because it does not have the weird logic. :-)